lib/model/message.dart

```diff
@@ -167,33 +168,61 @@ class MessageStoreImpl with MessageStore {
       return;
     }
 
-    if (newTopic == null) {
-      // The `subject` field (aka newTopic) is documented to be present on moves.
-      assert(debugLog('Malformed UpdateMessageEvent: move but no newTopic')); // TODO(log)
+    if (newTopic == null && newStreamId == null) {
+      // At least one of `subject` (aka newTopic) or `new_stream_id` (aka
+      // newStreamId) is documented to be present on moves.
+      assert(debugLog('Malformed UpdateMessageEvent: move but no newTopic or newStreamId')); // TODO(log)
       return;
     }
     if (origStreamId == null) {
       // The `stream_id` field (aka origStreamId) is documented to be present on moves.
       assert(debugLog('Malformed UpdateMessageEvent: move but no origStreamId')); // TODO(log)
       return;
     }
-
-    if (newStreamId == null
-        && MessageEditState.topicMoveWasResolveOrUnresolve(origTopic, newTopic)) {
-      // The topic was only resolved/unresolved.
-      // No change to the messages' editState.
+    if (propagateMode == null) {
+      // The `propagate_mode` field (aka propagateMode) is documented to be present on moves.
+      assert(debugLog('Malformed UpdateMessageEvent: move but no propagateMode')); // TODO(log)
       return;
     }
 
-    // TODO(#150): Handle message moves.  The views' recipient headers
-    //   may need updating, and consequently showSender too.
-    //   Currently only editState gets updated.
+    final wasResolveOrUnresolve = (newStreamId == null && newTopic != null
+      && MessageEditState.topicMoveWasResolveOrUnresolve(origTopic, newTopic));
```

Comment:
```suggestion
    final wasResolveOrUnresolve = (newStreamId == null
      && MessageEditState.topicMoveWasResolveOrUnresolve(origTopic, newTopic!));
```

The case where that `newTopic != null` condition would evaluate to false should be impossible, right? That's covered by the conditional above which leads to an early return.

When a condition is meant to be impossible, it's best to make that clear rather than just not do anything for that condition. Otherwise the reader has to wonder whether there's some handling that's missing for that condition.

Reply:
That's a good catch. Thanks!